### PR TITLE
Allow toggling of the dashboard subscription sidebar from the icon

### PR DESCRIPTION
**Pull request by kdoh**
### What
Updates the behavior of clicking the subscriptions icon in the dashboard header so that the sidebar gets toggled, instead of you having to go all the way down to `Cancel.` This now matches the behavior of the info panel.

### Demo
https://www.loom.com/share/465134c0f6ba4ffd9408a4dade4d9d67?sid=246a8af4-3112-4c88-b502-d8260d3de8cd

### Checklist

- [x] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: frontend/src/metabase/dashboard/components/Dashboard/Dashboard.jsx [hunk 1]
```diff
@@ -142,7 +142,7 @@ class DashboardInner extends Component {
   };
 
   onSharingClick = () => {
-    this.props.setSharing(true);
+    this.props.setSharing(this.props.isSharing ? false : true);
```
Comment (kulyk) [inline frontend/src/metabase/dashboard/components/Dashboard/Dashboard.jsx:145]: I think a more common way to implement "toggle" behavior is to do `setState(!currentState)`, like:

```js
this.props.setSharing(!this.props.isSharing)
```

File: e2e/test/scenarios/sharing/subscriptions.cy.spec.js [hunk 2]
```diff
@@ -66,6 +66,19 @@ describe("scenarios > dashboard > subscriptions", () => {
...
+  describe("sidebar toggling behavior", () => {
+    it("should allow toggling the sidebar", () => {
+      openDashboardSubscriptions();
+
+      cy.findByLabelText("subscriptions").click();
+      // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
+      cy.findByText("Email it").should("not.exist");
+      cy.findByLabelText("subscriptions").click();
+      // eslint-disable-next-line no-unscoped-text-selectors -- deprecated usage
+      cy.findByText("Email it");
```
Comment (kulyk) [inline e2e/test/scenarios/sharing/subscriptions.cy.spec.js:78]: Let's not use deprecated practices in new code. We expect `findByText` to be wrapped in `within` (more in #30059). But in this case, maybe we can assert that an `aside` disappears after we click the "subscriptions" button again? For example:

```js
sidebar().should("not.exist");
cy.findByLabelText("subscriptions").click();
sidebar().findByText("Email it").should("exist");
```

File: e2e/test/scenarios/sharing/subscriptions.cy.spec.js [hunk 3]
```diff
@@ -66,6 +66,19 @@ describe("scenarios > dashboard > subscriptions", () => {
...
+    });
```
Comment (kulyk) [inline e2e/test/scenarios/sharing/subscriptions.cy.spec.js:79]: We're doing this at the moment:

1. Open the subscriptions sidebar (with `openDashboardSubscriptions`)
2. Close it
3. Assert it disappeared
4. Open it again
5. Assert it appeared

Steps 4 and 5 seem redundant; we should be able to assert the sidebar is open after step 1

**Review by kulyk — COMMENTED** [review]
Looks good, inlined a couple of comments about the test

File: e2e/test/scenarios/sharing/subscriptions.cy.spec.js (hunk 3 above)
Comment (kdoh) [reply e2e/test/scenarios/sharing/subscriptions.cy.spec.js:79]: Fair. I guess I was going a bit overboard with "toggling" here and closing it after it's initially open via `openDashboardSubscriptions` does test the behavior.

File: e2e/test/scenarios/sharing/subscriptions.cy.spec.js (hunk 2 above)
Comment (kdoh) [reply e2e/test/scenarios/sharing/subscriptions.cy.spec.js:78]: Ah ok I see the better way to do it down below in the file, was looking at the tests up above. Easy to fix.

**Review by kulyk — APPROVED** [review]

**Pull request merged**